docker/package/package_generator.py

```diff
@@ -2,25 +2,57 @@
 #
 # SPDX-License-Identifier: LicenseRef-MIT-TQ
 
-import os, shutil, sys, subprocess, json
+import os, shutil, sys, subprocess, json, getopt
 from distutils.dir_util import copy_tree
 
 from .model import Package, print_service_file
 from .packages import packages
 
-if sys.argv[1] == "ubuntu":
-    is_ubuntu = True
-elif sys.argv[1] == "fedora":
-    is_ubuntu = False
-else:
-    raise Exception(f"Unexpected package target OS, only 'ubuntu' and 'fedora' are supported.")
+is_ubuntu = None
+is_source = None
+package_to_build = None
 
-if sys.argv[2] == "source":
-    is_source = True
-elif sys.argv[2] == "binary":
-    is_source = False
-else:
-    raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
+options, _ = getopt.getopt(sys.argv[1:], "", ['os=', 'type=', 'package='])
```

Comment:
It looks like bash-style of command line options parsing which is hard to read, error-prone and scales poorly. Can we use `argparse` instead?